testFlight1/lib/teensyShared/rocket.cpp

```diff
@@ -0,0 +1,244 @@
+                    break;
+                }
+ 
+                // measure the length of the coast time (for hysteresis)
+                rocketTimers.coast_timer =
+                    chVTGetSystemTime() - rocketTimers.burnout_time;
+ 
+                // If the low acceleration lasts long enough, coast is detected
+                if (rocketTimers.coast_timer > coast_time_thresh) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_COAST;
+                }
+ 
+            break;
+ 
+            case STATE_COAST:
+                //if velocity <= 0
+                if (pointer_struct->sensorDataPointer->state_data.state_vz <= apogee_thresh) {
+                    rocketTimers.apogee_time = chVTGetSystemTime();
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_APOGEE_DETECT;
+                }
+ 
+            break;
+ 
+            case STATE_APOGEE_DETECT:
+                //If the negative velocity was too brief, go back to coast
+                if (pointer_struct->sensorDataPointer->state_data.state_vz > apogee_thresh) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_COAST;
+                    break;
+                }
+ 
+                // measure the length of the apogee time
+                rocketTimers.apogee_timer =
+                    chVTGetSystemTime() - rocketTimers.apogee_time;
+ 
+                // If the < 0 velocity lasts long enough, apogee is detected
+                if (rocketTimers.coast_timer > apogee_time_thresh) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_APOGEE;
+                }
+            break;
+ 
+            case STATE_APOGEE: //*assumes velocity is <= 0
+                //If rocket has velocity in main range
+                if (pointer_struct->sensorDataPointer->state_data.state_vz < main_thresh_max
+                        && pointer_struct->sensorDataPointer->state_data.state_vz > main_thresh_min) {
+                    rocketTimers.main_time = chVTGetSystemTime();
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_MAIN_DETECT;
+                }
+                //If rocket has volocity in drogue range
+                if (pointer_struct->sensorDataPointer->state_data.state_vz < drogue_thresh) {
+                    rocketTimers.drogue_time = chVTGetSystemTime();
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_DROGUE_DETECT;
+                }
+                //If rocket has 0ish velocity
+                if (pointer_struct->sensorDataPointer->state_data.state_vz > landed_thresh) {
+                    rocketTimers.landing_time = chVTGetSystemTime();
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_LANDED_DETECT;
+                }
```

Comment:
So a few things stand out to me here.

1. I see you're looking to see if `state_vz` is within certain bounds to detect the main chute. Wouldn't it make more sense to look for "anything _slower_ than some threshold" to detect the main?

2. The "looking for `state_vz` within certain bounds" logic makes more sense for detecting drogue in my opinion.

3. While the "`state_vz` within bounds` logic isn't wrong here, it could make it possible to misunderstand the logic or accidentally reverse the min/max values. I'd suggest something like the following:
```
// in thresholds.h
const float kDrogueDescentRate = something;
const float kDrougeDescentRateEpsilon = something;

// in the FSM:
if (abs(state_vz - kDrogueDescentRate) < kDrogueDescentRateEpsilon) { ... }
```

4. Same thing as point 2 but for the landing detection.

Nits:
- I think it makes more sense to check for drogue conditions first, then main, the landed. Since that's what a nominal flight sequence will follow.

Reply:
I think your nit is actually a major change that should be made, especially with the current `if` instead of the `else if` conditions.

Reply:
Also I can't reply to your PR main comment @AyberkY so I'll just reply here. The current most up-to-date code is in the testFlight folder (which should never have been a separate folder at all, should've been a branch) so for this PR we're going to stick with it. As soon as this is over, I believe the acTeensy and hybridTeensy code is super outdated so I'll go ahead at delete those after a quick compare (you can look over too), and then we can rename the folder to be flightCode with subfolder platformio projects.